library/Fission/User/Types.hs

```diff
@@ -1,40 +1,36 @@
 module Fission.User.Types (User (..))where
 
 import Data.Swagger
-import Database.Selda
+import Fission.Prelude
 
-import qualified Fission.Platform.Heroku.AddOn as Heroku
+-- Database
 
-import Fission.Prelude
-import Fission.Security       (Digestable (..))
-import Fission.Security.Types (SecretDigest)
-import Fission.User.Role
+import           Fission.Storage.Persist (Generate(..))
+import qualified Fission.Storage.Persist as Persist
+
+-- User instances
 
 import qualified Fission.Internal.UTF8 as UTF8
+import           Fission.Security       (Digestable (..))
+import           Fission.Security.Types (SecretDigest)
+import           Fission.User.Role
+
+
+{-| This'll take our User model, defined in template Haskell,
+    and translate them into data and entity types automatically.
+-}
+Persist.generate
+  [ Types ]
+  [ "library/Fission/User/Model" ]
 
--- | A user account, most likely a developer
-data User = User
-  { userID        :: ID User
-  , username      :: Text
-  , email         :: Maybe Text
-  , role          :: Role
-  , active        :: Bool
-  , herokuAddOnId :: Maybe (ID Heroku.AddOn)
-  , secretDigest  :: SecretDigest
-  , insertedAt    :: UTCTime
-  , modifiedAt    :: UTCTime
-  } deriving ( Show
-             , Eq
-             , Generic
-             , SqlRow
-             )
-
-instance Digestable (ID User) where
-  digest = digest . UTF8.textShow
-
-instance ToSchema (ID User) where
-  declareNamedSchema _ =
-    mempty
-      |> type_ ?~ SwaggerInteger
-      |> NamedSchema (Just "UserID")
-      |> pure
+-- TODO: Not sure what to do with this yet.
+--
+-- instance Digestable (ID User) where
+--   digest = digest . UTF8.textShow
+--
+-- instance ToSchema (ID User) where
+--   declareNamedSchema _ =
+--     mempty
+--       |> type_ ?~ SwaggerInteger
+--       |> NamedSchema (Just "UserID")
+--       |> pure
```

Comment:
🤔 Without having tried it, my assumption is to use the primary key:

```haskell
instance ToSchema (UserId) where
  declareNamedSchema _ = -- probably the same as before
```

Reply:
If you want to go down the rabbit hole, we may be able to do this generically. From Persistent:

```haskell
class ( PersistField (Key record), ToJSON (Key record), FromJSON (Key record)
      , Show (Key record), Read (Key record), Eq (Key record), Ord (Key record))
  => PersistEntity record where
```

With that in mind:

```haskell
-- maybe need this constraint? No sure: PersistField (Key model) => ...
instance ToSchema (Key model) where
  declareNamedSchema proxy =
    mempty
      |> type_ ?~ SwaggerInteger
      |> NamedSchema (Just (recordName proxy <> "Id"))
      |> pure
```

Buuuuuuuuuuuuuut that may be more complicated 🤷‍♀️ It's flexible at least

DISCLAIMER: I haven't tried that code, so YMMV